book/src/programmability/witness-pattern.md

```diff
@@ -0,0 +1,277 @@
+/// May or may not be implemented by the application.
+public fun transfer<T: store + drop>(_metadata: T, ticket: Ticket<T>, to: address) {
+    transfer::transfer(ticket, to)
+}
+```
+
+In the example above, the instantiation of the `Ticket` in the `new` function is straightforward, as
+the metadata is passed as an argument. However, the `transfer` function, which can be implemented by
+the application, requires some kind of authorization. And the only authorization method we have here
+is the metadata itself. So, the application can pass an empty metadata struct as a Witness to the
+`transfer` function, hence proving that it has the right to transfer the ticket of this type.
+
+```move
+module book::concert {
+    use book::ticket::{Self, Ticket};
+
+    /// The metadata for the concert ticket.
+    public struct Metadata has store, drop {
+        name: String,
+        date: String,
+        event_id: u64
+    }
+
+    /// Purchase a ticket for the concert. (omits the actual purchase logic)
+    public fun purchase(/* take a Coin */ ctx: &mut TxContext) {
+        let ticket = ticket::new(Metadata {
+            name: b"Mysties".to_string(),
+            date: b"2023-05-03".to_string(),
+            event_id: 1
+        });
+
+        // Transfer the ticket to the user.
+        ticket::transfer(empty(), ticket, ctx.sender())
+    }
+
+    /// Create an empty Metadata struct to act as a Witness.
+    fun empty(): Metadata {
+        Metadata { name: b"".to_string(), date: b"".to_string(), event_id: 0 }
+    }
+}
+```
+
+## Limitations
+
+- Witnesses are a powerful tool, but they have their limitations. The most important one is that
+  they do not guarantee uniqueness. The module owning the `T` can create multiple witnesses of the
+  same type, and if the library / module functionality assumes that the witness is unique and
+  instantiated only once, it may lead to unexpected behaviour. To address this issue, the
+  [One Time Witness](./one-time-witness.md) pattern is introduced, which ensures that the witness is
+  created only once.
+
+- In many cases Witness requires a function to be implemented in the instantiating module. This can
+  be seen as a limitation, as it requires additional boilerplate code to be written. However, it is
+  not the case for [One Time Witness](./one-time-witness.md), and in the cases where a regular
+  Witness can be used, the Witness can be created in the [module `init`](./module-initializer.md)
+  function.
+
+## Summary
+
+- Witness is a type with the `drop` ability, used to instantiate generic types with a concrete type
+  parameter.
+- It acts as a proof of ownership of a type and may be used for type-based authorization.
+- Witness can be also used to create abstract class-like behaviors in Move.
+- Transferable Witness is a variation of the Witness that has the `store` ability, allowing it to be
+  stored in objects. It may be useful in certain applications, but not encouraged.
```

Comment:
If the holder of smart contract gates the Transferable Witness with an `AdminCap`, then would this be considered safe as no one else can access nor modify the Transferable Witness?